processor/lsmintervalprocessor/internal/merger/model.go

```diff
@@ -406,3 +417,126 @@ func mergeDeltaHistogramDP(from, to pmetric.HistogramDataPoint) {
 		toCounts.SetAt(i, fromCounts.At(i)+toCounts.At(i))
 	}
 }
+
+func mergeDeltaExponentialHistogramDP(from, to pmetric.ExponentialHistogramDataPoint) {
+	if to.Count() == 0 {
+		// Take min max from `from` as `to` is empty
+		to.SetMin(from.Min())
+		to.SetMax(from.Max())
+	} else if from.Count() != 0 {
+		// If both `to` and `from` are not empty then calculate
+		to.SetMin(min(to.Min(), from.Min()))
+		to.SetMax(max(to.Max(), from.Max()))
+	}
+
+	to.SetSum(to.Sum() + from.Sum())
+	to.SetCount(to.Count() + from.Count())
+	// TODO (lahsivjar): Zero count depends on the value of zero threshold.
+	// If zero threshold are not the same then we can't merge them directly,
+	// instead, we will need to expand the zero threshold to be inclusive
+	// of the smaller zero count. If a zero threshold encroaches a bucket,
+	// then we will need to expand the zero count till the end of the bucket.
+	to.SetZeroCount(to.ZeroCount() + from.ZeroCount())
```

Comment:
[**For reviewers**] Will do this in a followup.

Reply:
FYI there's logic you could copy at https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/deltatocumulativeprocessor/internal/data/expo/zero.go

I wonder if we should just vendor and use that code, under the assumption that lsmintervalprocessor can be moved to opentelemetry-collector-contrib. At that point we could create a common (internal?) package for merging exponential histograms -- or maybe metrics in general.

Reply:
> FYI there's logic you could copy

Ah, thanks! This would be helpful.

> I wonder if we should just vendor and use that code

I don't think we can. Looked into the code briefly and it only merges buckets of same scale: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/a1915508175a498b731d17c951d776761ce8e824/processor/deltatocumulativeprocessor/internal/data/expo/merge.go#L10-L12

Maybe I will try to upstream the changes in this PR to support merges across scale.

Reply:
> Maybe I will try to upstream the changes in this PR to support merges across scale.

Just realized that it doesn't make sense for deltatocumulative to support multiple scales. Maybe we can create a common exphistogram package in future when we try to upstream this.

Reply:
@lahsivjar it's true that the Merge function only works with buckets of the same scale, but there's also code to downscale: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/a1915508175a498b731d17c951d776761ce8e824/processor/deltatocumulativeprocessor/internal/data/expo/scale.go#L42

They're all put together here: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/a1915508175a498b731d17c951d776761ce8e824/processor/deltatocumulativeprocessor/internal/data/add.go#L68-L110

Reply:
Hmm, the logic is very similar but I am a bit reluctant since we will have to vendor the code and build `Merge` on top of it.

Reply:
I think the collapse logic is a bit complicated upstream in terms of how it is written - nothing major. It seems I am getting attached to my code 😅 . It might be better to just vendor. Let me update!

Reply:
Updated the code to vendor the internal package. Will also update histogram to use that instead in a separate PR.